### PR TITLE
refactor(scroll-list): remove `watchScrollWindowSize` prop of ScrollList

### PR DESCRIPTION
**Pull request by plisovyi**
in favor of always watching scrollWindow element or Window for resize

### CODE REVIEW

File: packages/components/src/common/dimensions.ts [hunk 1]
```diff
@@ -16,3 +16,69 @@ export const unmeasuredDimensions: ElementDimensions = {
 export interface DimensionsById {
     [id: string]: ElementDimensions;
 }
+
+export const observeElementDimensions = (element: HTMLElement, callback: (dimensions: ElementDimensions) => void) => {
+    const observer = new ResizeObserver(() => callback(element.getBoundingClientRect()));
+
+    if (element !== null) return;
```
Comment (VladyslavG) [inline packages/components/src/common/dimensions.ts:23]: I don't get this line

File: packages/components/src/common/dimensions.ts [hunk 2]
```diff
@@ -16,3 +16,69 @@ export const unmeasuredDimensions: ElementDimensions = {
...
+
+    window.addEventListener('resize', onResize, {
+        capture: false,
+        passive: true,
+    });
+
+    return () => window.removeEventListener('resize', onResize);
+};
+
+export const getSizeFromDimensions = (dimensions: ElementDimensions, sizeAsHeight: boolean): number => {
```
Comment (VladyslavG) [inline packages/components/src/common/dimensions.ts:45]: I don't like the name sizeAsHeight

File: packages/components/src/common/dimensions.ts [hunk 3]
```diff
@@ -16,3 +16,69 @@ export const unmeasuredDimensions: ElementDimensions = {
...
+export const getDimensionsFromRect = (rect?: DOMRect): ElementDimensions => {
+    return rect
+        ? {
+              width: rect.width,
+              height: rect.height,
+          }
+        : unmeasuredDimensions;
+};
+
+export const getElementDimensions = (element: HTMLElement | null | undefined): ElementDimensions => {
```
Comment (VladyslavG) [inline packages/components/src/common/dimensions.ts:58]: so if I pass null or undefined , I get window dimensions ?
it is not expected from function name

File: packages/components/src/common/dimensions.ts [hunk 4]
```diff
@@ -16,3 +16,69 @@ export const unmeasuredDimensions: ElementDimensions = {
...
+    return element
+        ? getDimensionsFromRect(element.getBoundingClientRect())
+        : {
+              width: window.innerWidth,
+              height: window.innerHeight,
+          };
+};
+
+/**
+ * Get Element or Window size
```
Comment (VladyslavG) [inline packages/components/src/common/dimensions.ts:68]: but this comment contradicts name of function

File: packages/components/src/hooks/use-element-dimensions.ts [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-} from '../hooks';
+    childrenById,
+    DimensionsById,
+    ElementDimensions,
+    getDimensionsFromRect,
+    unmeasuredDimensions,
+} from '../common';
+import { unchanged, useDelayedUpdateState } from '../hooks';
 
 const elementDimensions = (element?: Element): ElementDimensions => {
```
Comment (VladyslavG) [inline packages/components/src/hooks/use-element-dimensions.ts:11]: this name should include `get`

File: packages/components/src/hooks/use-element-size.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import type React from 'react';
+import { useEffect, useState } from 'react';
+import { getSizeFromDimensions, getElementSize, observeElementDimensions, observeWindowDimensions } from '../common';
+
+export const useElementSize = (element: React.RefObject<HTMLElement> | undefined, sizeAsHeight: boolean): number => {
```
Comment (VladyslavG) [inline packages/components/src/hooks/use-element-size.ts:5]: this is not expected that function with such name will observe window dimensions

File: packages/components/src/common/dimensions.ts (hunk 4 above)
Comment (plisovyi) [reply packages/components/src/common/dimensions.ts:70]: Discussed, will be executed later in stylable-components

File: packages/components/src/hooks/use-element-size.ts (hunk 6 above)
Comment (plisovyi) [reply packages/components/src/hooks/use-element-size.ts:5]: Discussed, will be executed later in stylable-components

File: packages/components/src/common/dimensions.ts (hunk 3 above)
Comment (plisovyi) [reply packages/components/src/common/dimensions.ts:60]: Discussed, will be executed later in stylable-components

File: packages/components/src/common/dimensions.ts (hunk 2 above)
Comment (plisovyi) [reply packages/components/src/common/dimensions.ts:47]: Discussed, will be executed later in stylable-components

File: packages/components/src/common/dimensions.ts (hunk 1 above)
Comment (plisovyi) [reply packages/components/src/common/dimensions.ts]: Thank you for catching it! Fixed 🤓

File: packages/components/src/hooks/use-element-size.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    useEffect(() => {
+        const cleanup = element?.current
+            ? observeElementDimensions(element.current, (dimensions) =>
+                  updateSize(getSizeFromDimensions(dimensions, sizeAsHeight))
+              )
+            : observeWindowDimensions((dimensions) => updateSize(getSizeFromDimensions(dimensions, sizeAsHeight)));
+
+        updateSize(getElementSize(element?.current, sizeAsHeight));
+
+        return () => cleanup();
```
Comment (VladyslavG) [inline packages/components/src/hooks/use-element-size.ts:17]: you can simply

`return cleanup;` if you like such style

File: packages/components/src/scroll-list/scroll-list.tsx [hunk 8]
```diff
@@ -171,18 +162,16 @@ export function ScrollList<T, EL extends HTMLElement = HTMLDivElement>({
...
     const listRef = (listRoot?.props?.ref as React.RefObject<HTMLDivElement>) || defaultListRef;
-    /**
-     * get scroll position of scrollWindow and trigger re-rendering on its change
-     */
+    /** get scrollWindow's or Window's scroll position and trigger re-rendering on its change */
     const scrollPosition = useScroll({
         isHorizontal,
         ref: scrollWindow,
     });
+    /** get scrollWindow's or Window's size and trigger re-rendering on its change */
```
Comment (VladyslavG) [inline packages/components/src/scroll-list/scroll-list.tsx:170]: don't you think that this comment is kind of obvious if you know how react hooks work ?

Comment (VladyslavG) [reply packages/components/src/scroll-list/scroll-list.tsx:170]: at lease second part about rerendering

Comment (plisovyi) [reply packages/components/src/scroll-list/scroll-list.tsx:170]: Thought about it as more of reminder of scrollWindow/Window conundrum 🤷🏼

**Review by VladyslavG — APPROVED** [review]

**Pull request merged**